### ISSUE TITLE
Focus and content issues with list block

### ISSUE THREAD

**Issue by nylen** [WordPress/gutenberg#567]
In `master` I'm seeing some pretty severe issues with the list block:

- It only accepts the focus when clicking **outside** of the list items' text, **not** when clicking on a word
- Whenever any other Redux action is fired (the block is re-rendered?), any edits are discarded.  Simply moving the mouse over the block is enough to trigger this.

Here's a screen recording of me attempting to perform some basic actions with the list block:

![gutenberg-list-bugginess](https://cloud.githubusercontent.com/assets/227022/25547055/86a8f38a-2c33-11e7-90e3-8441fc3258ae.gif)

Others in chats have been unable to reproduce this issue, I'm not sure if that's because it's just really strange or because it's only happening on my computer.  I'm using Chrome 57.0.2987.110 (64-bit) on Debian-based Linux.

In any case, it looks like there is some still-undiscovered major bugginess here.  A few past issues/PRs look pretty closely related:

- #480 (in particular I suspect a possible connection between the `setTimeout` here and the fact that the focus jumps briefly to the list block, then back - cc @youknowriad)
- #501 and #502 (the opposite issue, where it was not possible to un-focus the list block)
- #529 (looks related to the content-disappearing issues, but **does not** fix them for me)

**Comment by EphoxJames** [comment]
Some of these were to be addressed by the pull request
https://github.com/WordPress/gutenberg/pull/396
Unfortunately mimo84 went on holidays before all the issues could be resolved and we didn't have a chance to do a proper hand-off. I'll be working on bringing that back up to speed with the current state of the repo today.

**Comment by afercia** [comment]
Also, when navigating through blocks using only the Tab key, the list block is a keyboard trap. I guess because the Tab key is supposed to be used for the list items indentation (nesting) so at each Tab key press (or Shift+Tab) nothing happens and it's impossible to navigate away from this block using only the keyboard.
This would be a serious accessibility blocker.

**Comment by nylen** [comment]
This appears to have been fixed by an unrelated PR.  I'm not sure which one, but this is a bit worrying.  If there was a PR that addressed this issue specifically then it should have been linked to this issue.  Anyway I'll close this out.  @afercia the Tab keyboard trap should be a separate issue.
